### PR TITLE
Add support for PVCs

### PR DESCRIPTION
**Pull request by mohabusama**
Add discovery of kubernetes Persistent Volume Claims and their supporting backend.

+ Add AWSElasticBlockStorage volume support

### CODE REVIEW

File: zmon_agent/discovery/kubernetes/cluster.py [hunk 1]
```diff
@@ -647,6 +653,52 @@ def get_cluster_ingresses(kube_client, cluster_id, alias, environment, region, i
...
 
+@trace(tags={'kubernetes': 'persistentvolumeclaim'}, pass_span=True)
+def get_cluster_persistentvolumeclaims(kube_client, cluster_id, alias, environment, region, infrastructure_account,
+                                       namespace='default', **kwargs) -> list:
+    current_span = extract_span_from_kwargs(**kwargs)
+
+    entities = []
+
+    pvcs = get_all(kube_client, kube_client.get_persistentvolumeclaims, namespace, span=current_span)
+    pvs = get_all(kube_client, kube_client.get_persistentvolumes, namespace, span=current_span)
```
Comment (vetinari) [inline zmon_agent/discovery/kubernetes/cluster.py:664]: is this used somewhere?

File: zmon_agent/discovery/kubernetes/volumes.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    fields['storage_size'] = pv.obj['spec'].get('capacity', {}).get('storage')
+
+    return fields
+
+
+def get_aws_ebs_fields(pv: PersistentVolume) -> dict:
+    """Return volume fields for AWSElasticBlockStore plugin"""
+
+    spec = pv.obj['spec']['awsElasticBlockStore']
```
Comment (mikkeloscar) [inline zmon_agent/discovery/kubernetes/volumes.py:30]: Can this break if someone create a pv without this key?

File: zmon_agent/discovery/kubernetes/cluster.py (hunk 1 above)
Comment (mohabusama) [reply zmon_agent/discovery/kubernetes/cluster.py:664]: a few lines down ;)

File: zmon_agent/discovery/kubernetes/volumes.py (hunk 2 above)
Comment (mikkeloscar) [reply zmon_agent/discovery/kubernetes/volumes.py:30]: Or you do some fancy function mapping, I see :)

File: zmon_agent/discovery/kubernetes/cluster.py (hunk 1 above)
Comment (vetinari) [reply zmon_agent/discovery/kubernetes/cluster.py:664]: ouch, yes  :)

**Comment by vetinari** [comment]
:+1:

**Comment by mikkeloscar** [comment]
:+1:

**Comment by mohabusama** [comment]
Added implementation for ``TimedHTTPClient`` as there are no timeouts set for ``pykube.HTTPClient``

PR upstream: https://github.com/kelproject/pykube/pull/156

**Comment by elauria** [comment]
👍

**Comment by lmineiro** [comment]
:+1:

**Pull request merged**